server/src/main/java/org/opensearch/index/TieredMergePolicyProvider.java

```diff
@@ -136,7 +136,13 @@ public final class TieredMergePolicyProvider implements MergePolicyProvider {
     private final boolean mergesEnabled;
 
     public static final double DEFAULT_EXPUNGE_DELETES_ALLOWED = 10d;
-    public static final ByteSizeValue DEFAULT_FLOOR_SEGMENT = new ByteSizeValue(2, ByteSizeUnit.MB);
+
+    /**
+     *  Using 16MB floor segment size rather than the original defaults in Lucene branch_10x.
+     *  The 16MB floor change was planned to be included in Lucene 10.2 but we're adopting it early to gain the performance benefits of larger floor segments.
```

Comment:
Sorry, nitpicking on the comments. The "was planned to be included..." sounds like this change didn't happen in Lucene, but the change has been made and will be included in the next release: https://github.com/apache/lucene/pull/14189

Also, the way our code is structured is that I think we never fallback to the Lucene defaults because we have our own default value here that is always applied. All this is to say is that we can probably just say:

```
Use 16MB floor size to match Lucene default.
See https://github.com/apache/lucene/pull/14189
```

Reply:
Sure Andrew, I just want to make sure we are not making any drastic change (drifting away) from Lucene. Since for OpenSearch we are going with 10.1.0 of Lucene and 16MB floor size change for Lucene is planned for 10.2.0 I have added some extra story.

Updated the PR, please check. Thanks

Reply:
> 16MB floor size change for Lucene is planned for 10.2.0 I have added some extra story.

I think it's only on the `main` (Lucene 11) branch.

Reply:
Seems like it's there on `branch_10x`: https://github.com/apache/lucene/blob/branch_10x/lucene/core/src/java/org/apache/lucene/index/LogByteSizeMergePolicy.java#L28